hack/compare_scc.sh

```diff
@@ -19,31 +19,46 @@
 
 set -ex
 
-SCC_BEFORE_SUFFIX=.yaml.before
-SCC_AFTER_SUFFIX=.yaml.after
-
-function dump_sccs_before(){
-    if [ "${CMD}" == "oc" ]; then
-        mkdir -p _out/scc
-        for SCCNAME in $( ${CMD} get scc -o custom-columns=:metadata.name )
-        do
-          echo -e "\n--- SCC ${SCCNAME} ---"
-          ${CMD} get scc ${SCCNAME} -o yaml > "_out/scc/${SCCNAME}${SCC_BEFORE_SUFFIX}" || true
-        done
-    else
-        echo "Ignoring SCCs on k8s"
-    fi
+OUTPUT_DIR=${OUTPUT_DIR:-_out}
+SCC_OUTPUT_DIR="${OUTPUT_DIR}/scc"
+
+BEFORE_DIR="${SCC_OUTPUT_DIR}/before"
+AFTER_DIR="${SCC_OUTPUT_DIR}/after"
+
+SUFFIX=.json
+
+function dump_sccs_before() {
+  if [ "${CMD}" == "oc" ] && [ "${KUBEVIRT_PROVIDER}" != "external" ]; then
+    mkdir -p "${BEFORE_DIR}"
+    for SCCNAME in $( ${CMD} get scc -o custom-columns=:metadata.name); do
+      echo -e "\n--- SCC ${SCCNAME} ---"
+      ${CMD} get scc "${SCCNAME}" -o json | jq 'del(.metadata.resourceVersion,.metadata.generation,.metadata.labels,.metadata.annotations)' > "${BEFORE_DIR}/${SCCNAME}${SUFFIX}" || true
+    done
+  else
+    echo "Ignoring SCCs on k8s"
```

Comment:
if `${CMD} == "oc"` **and**  `"${KUBEVIRT_PROVIDER}" != "external"` (e.g. CRC), it can still be OCP and not plain kubernetes.

Reply:
It *should* be OCP and not plain K8s

Reply:
yes, but in that case, the `if` statement is not satisfied and we're at: `echo "Ignoring SCCs on k8s"`, although we can be on OCP (CRC), which has SCCs